src/openloco/map/tilemgr.cpp

```diff
@@ -182,7 +187,19 @@ namespace openloco::map::tilemgr
     // 0x004610F2
     void map_invalidate_selection_rect()
     {
-        call(0x004610F2);
+        if ((input::getMapSelectionFlags() & 1) != 0)
+        {
+            for (uint16_t x = _mapSelectionAX; x <= _mapSelectionBX; x += 32)
+            {
+                for (uint16_t y = _mapSelectionAY; y <= _mapSelectionBY; y += 32)
+                {
+                    registers regs;
+                    regs.ax = x;
+                    regs.cx = y;
+                    call(0x004CBE5F, regs);
```

Comment:
Can you take this out into a separate function. 0x4CBE5F will be something like a invalidate map tile function.